packages/e2e-tests/specs/experiments/widgets.test.js

```diff
@@ -0,0 +1,363 @@
+			'//*[@role="option"][*[text()="Paragraph"]]'
+		);
+
+		return addParagraphBlock;
+	}
+
+	async function expectInsertionPointIndicatorToBeBelowLastBlock(
+		widgetArea
+	) {
+		const childBlocks = await widgetArea.$$( '[data-block]' );
+		const lastBlock = childBlocks[ childBlocks.length - 1 ];
+		const lastBlockBoundingBox = await lastBlock.boundingBox();
+
+		// TODO: Probably need a more accessible way to select this, maybe a test ID or data attribute.
+		const insertionPointIndicator = await page.$(
+			'.block-editor-block-list__insertion-point-indicator'
+		);
+		const insertionPointIndicatorBoundingBox = await insertionPointIndicator.boundingBox();
+
+		expect(
+			insertionPointIndicatorBoundingBox.y > lastBlockBoundingBox.y
+		).toBe( true );
+	}
+
+	it( 'Should insert content using the global inserter', async () => {
+		const widgetAreas = await page.$$(
+			'[aria-label="Block: Widget Area"][role="group"]'
+		);
+		const [ firstWidgetArea ] = widgetAreas;
+
+		let addParagraphBlock = await getParagraphBlockInGlobalInserter();
+		await addParagraphBlock.hover();
+
+		// FIXME: The insertion point indicator is not showing when the widget area has no blocks.
+		// await expectInsertionPointIndicatorToBeBelowLastBlock(
+		// 	firstWidgetArea
+		// );
+
+		await addParagraphBlock.click();
+
+		const addedParagraphBlockInFirstWidgetArea = await firstWidgetArea.$(
+			'[data-block][data-type="core/paragraph"][aria-label^="Empty block"]'
+		);
+
+		expect(
+			await addedParagraphBlockInFirstWidgetArea.evaluate(
+				( node ) => node === document.activeElement
+			)
+		).toBe( true );
+
+		await page.keyboard.type( 'First Paragraph' );
+
+		addParagraphBlock = await getParagraphBlockInGlobalInserter();
+		await addParagraphBlock.hover();
+
+		await expectInsertionPointIndicatorToBeBelowLastBlock(
+			firstWidgetArea
+		);
+		await addParagraphBlock.click();
+
+		await page.keyboard.type( 'Second Paragraph' );
+
+		/**
+		 * FIXME: There seems to have a bug when saving the widgets
+		 */
+		// await secondWidgetArea.click();
+
+		// addParagraphBlock = await getParagraphBlockInGlobalInserter();
+		// await addParagraphBlock.hover();
+
+		// // FIXME: The insertion point indicator is not showing when the widget area has no blocks.
+		// // await expectInsertionPointIndicatorToBeBelowLastBlock(
+		// // 	secondWidgetArea
+		// // );
+
+		// await addParagraphBlock.click();
+
+		// const addedParagraphBlockInSecondWidgetArea = await secondWidgetArea.$(
+		// 	'[data-block][data-type="core/paragraph"][aria-label^="Empty block"]'
+		// );
+
+		// expect(
+		// 	await addedParagraphBlockInSecondWidgetArea.evaluate(
+		// 		( node ) => node === document.activeElement
+		// 	)
+		// ).toBe( true );
+
+		// await page.keyboard.type( 'Third Paragraph' );
+
+		await saveWidgets();
+		const serializedWidgetAreas = await getSerializedWidgetAreas();
+		expect( serializedWidgetAreas ).toMatchInlineSnapshot( `
```

Comment:
We so rarely use _inline_ snapshots which is a shame because it's much nicer to review!

Reply:
Yeah! What's up with that! Inline snapshots rock!

Reply:
Probably because we didn't use to use prettier which helps a lot.

Reply:
I thought jest bundles its own prettier to do the formatting itself so we won't have to worry about that. I think it's simply because it's a less known feature 😆 .